domain-models-runtime/src/main/java/org/folio/rest/persist/PgUtil.java

```diff
@@ -397,53 +398,71 @@ static Method respond422method(Class<? extends ResponseDelegate> clazz) {
    * @param clazz  the ResponseDelegate class created from the RAML file with these methods:
    *               respond204(), respond400WithTextPlain(Object), respond404WithTextPlain(Object),
    *               respond500WithTextPlain(Object).
-   * @param asyncResultHandler  where to return the result created by clazz
+   * @return future where to return the result created by clazz
    */
-  public static void deleteById(String table, String id,
+   public static Future<Response> deleteById(String table, String id,
       Map<String, String> okapiHeaders, Context vertxContext,
-      Class<? extends ResponseDelegate> clazz,
-      Handler<AsyncResult<Response>> asyncResultHandler) {
-
-    final Method respond500;
+      Class<? extends ResponseDelegate> clazz) {
+        final Method respond500;
     try {
       respond500 = clazz.getMethod(RESPOND_500_WITH_TEXT_PLAIN, Object.class);
     } catch (Exception e) {
       logger.error(e.getMessage(), e);
-      asyncResultHandler.handle(response(e.getMessage(), null, null));
-      return;
+      return response(e.getMessage(), null, null);
     }
 
     try {
       Method respond204 = clazz.getMethod(RESPOND_204);
       Method respond400 = clazz.getMethod(RESPOND_400_WITH_TEXT_PLAIN, Object.class);
       Method respond404 = clazz.getMethod(RESPOND_404_WITH_TEXT_PLAIN, Object.class);
       if (! UuidUtil.isUuid(id)) {
-        asyncResultHandler.handle(responseInvalidUuid(table + ".id", id, clazz, respond400, respond500));
-        return;
+        return responseInvalidUuid(table + ".id", id, clazz, respond400, respond500);
       }
+
+      Promise<Response> promise = Promise.promise();
+
       PostgresClient postgresClient = PgUtil.postgresClient(vertxContext, okapiHeaders);
       postgresClient.delete(table, id, reply -> {
         if (reply.failed()) {
-          asyncResultHandler.handle(response(table, id, reply.cause(), clazz, respond400, respond500));
-          return;
+          response(table, id, reply.cause(), clazz, respond400, respond500).onComplete(promise);
```

Comment:
return;

is missing here and at similar places below.

Reply:
Thanks!